### DISCUSSION TITLE
Easy to specifying JVM memory settings such as heap min/max sizes

### DISCUSSION THREAD

**Discussion by davsclaus** [jbangdev/jbang#1441] [Q&A] [answered]
I have looked in the documentation, jbang --help, and searched a bit on the internet, but could not find out if JBang offered a nice and easy way to specify JVM memory settings when running a script.

At Apache Camel we love JBang and are using it for Camel CLI and a new way to use Camel. 

In the Camel CLI we made a command to output status of the running Camel integrations and one shows some memory and other status, eg as shown below:

```
  PID   NAME        JAVA     CAMEL            PLATFORM  STATUS    AGE         HEAP       NON-HEAP      GC      THREADS   CLASSES
 56699  bar.yaml    11.0.13  3.19.0-SNAPSHOT  JBang     Running  12h27m  76/322/4294 MB  59/62 MB  215ms (20)    7/9    6124/6124
 56712  chuck.yaml  11.0.13  3.19.0-SNAPSHOT  JBang     Running  12h27m  48/322/4294 MB  82/86 MB  317ms (39)    8/9    7480/7480
```

The heap stats are: used/committed/max
If you look then they all have 4gb as max. 

I wonder if there is a way to set the min/max size ala `-Xms64m -Xmx512m` but maybe in an easier way? And also if these could be set in the script file itself, like you have //DEPS and whatnot (https://github.com/apache/camel/blob/main/dsl/camel-jbang/camel-jbang-main/dist/CamelJBang.java#L22).

**Comment by quintesse** [comment, marked as answer]
Hi @davsclaus , yes you can set compile and runtime option sin the script using `//JAVAC_OPTIONS` and `//JAVA_OPTIONS`, see https://www.jbang.dev/documentation/guide/latest/running.html#java-and-javac-options

And you can also use the command line flag `--java-options` for passing runtime options.

**Comment by davsclaus** [comment]
Thanks for the pointers, this is how you can do it

```
jbang run --java-options=-Xmx300m camel@apache/camel run bar.yaml
```

Setting heap sizes is more common to do, but can be a bit tricky to do. 
Maybe add an example in the docs, at that link you provided.

**Comment by davsclaus** [comment]
Also the `--java-options` is not possible for "app" like we have with camel, so eg you cannot do

camel --java-options=-Xmx300m run bar.yaml

I don't think I have been able to add `--java-options` in our custom Camel JBang CLI - eg its the same when I need to enable JVM debug, then I must do

jbang run --debug camel@apache/camel run bar.yaml

**Reply by quintesse** [reply]
Indeed, right now we don't have an option to pass arguments to Jbang when using an installed "app".
The only way you could make that work is using the `JBANG_JAVA_OPTIONS` env var, like this (didn't test, but should work):

```
JBANG_JAVA_OPTIONS=-Xmx300m camel run bar.yaml
```

**Reply by quintesse** [reply]
I opened an issue for this: https://github.com/jbangdev/jbang/issues/1442
